src/handlers/profile/mod.rs

```diff
@@ -1,2 +1,87 @@
+use {
+    ethers::types::H160,
+    serde::{Deserialize, Serialize},
+    std::str::FromStr,
+};
+
 pub mod lookup;
+pub mod register;
 pub mod reverse;
+
+/// Payload to register domain name that should be serialized to JSON
+/// and passed to the RegisterRequest.message
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct RegisterPayload {
+    /// Name to register
+    pub name: String,
+    /// Address
+    pub address: String,
+    /// Unixtime
+    pub timestamp: u64,
+}
+/// Data structure representing a request to register a name
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct RegisterRequest {
+    /// Serialized JSON register payload
+    pub message: String,
+    /// Message signature
+    pub signature: String,
+    /// Address
+    pub address: String,
+}
+
+#[tracing::instrument]
+pub fn verify_message_signature(
+    message: &str,
+    signature: &str,
+    owner: &H160,
+) -> Result<bool, Box<dyn std::error::Error>> {
+    let prefixed_message = format!("\x19Ethereum Signed Message:\n{}{}", message.len(), message);
+    let message_hash = ethers::core::utils::keccak256(prefixed_message.clone());
+    let message_hash = ethers::types::H256::from_slice(&message_hash);
+
+    let sign = ethers::types::Signature::from_str(signature)?;
+    match sign.verify(message_hash, *owner) {
+        Ok(_) => Ok(true),
+        Err(_) => Ok(false),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use {super::*, ethers::types::H160, std::str::FromStr};
+
+    #[test]
+    fn test_verify_message_signature_valid() {
+        let message = "test message signature";
+        let signature = "0x660739ee06920c5f55fbaf0da4f435faaa9c55e2c9da303c50c4b3865191d67e5002a0b10eb0f89bae66823f7f07415ea9d5bbb607ee61ac98b7f2a0a44fcb5c1b";
+        let owner = H160::from_str("0xAff392551773CCb2574fAE23195CC3aFDBe98d18").unwrap();
+
+        let result = verify_message_signature(message, signature, &owner);
+        assert!(result.is_ok());
+        assert!(result.unwrap());
+    }
+
+    #[test]
+    fn test_verify_message_signature_json() {
+        let message = "{\\\"test\\\":\\\"some my text\\\"}";
```

Comment:
Raw strings can help here with the escape confusion:

```suggestion
        let message = r#"{\"test\":\"some my text\"}"#;
```

Also, shouldn't this be updated to not have the escapes?

Reply:
I don't mind changing it to the raw string to remove the ugly `\\\`.

> Also, shouldn't this be updated to not have the escapes? Guess it doesn't really matter because this is a generic signature validation test.

Yes, it doesn't matter what the signed string looks like, but as we are using the escaped JSON in our checks let's stick to it on test as well.

Thanks for the comments!